expression/builtin_cast_vec.go

```diff
@@ -698,11 +700,31 @@ func (b *builtinCastTimeAsDurationSig) vecEvalDuration(input *chunk.Chunk, resul
 }
 
 func (b *builtinCastDurationAsDurationSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastDurationAsDurationSig) vecEvalDuration(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	var err error
+	if err = b.args[0].VecEvalDuration(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	res := result.GoDurations()
+	var fsp int8
+	fsp, err = types.CheckFsp(b.tp.Decimal)
+	if err != nil {
+		return err
+	}
+	var zeroT time.Time
+	for i, v := range res {
+		if result.IsNull(i) {
+			continue
+		}
+		zeroT = time.Date(0, 0, 0, 0, 0, 0, 0, time.Local)
+		nd := zeroT.Add(v).Round(time.Duration(math.Pow10(9-int(fsp))) * time.Nanosecond).Sub(zeroT)
```

Comment:
Extracting code from `Duration.RoundFrac` to here seems dangerous.
How about:
```
	dur := &types.Duration{
		Fsp: types.UnspecifiedFsp,
	}
	for i, v := range res {
		if result.IsNull(i) {
			continue
		}
		dur.Duration = v
		rd, err := dur.RoundFrac(int8(b.tp.Decimal))
		// handle this err
		res[i] = rd.Duration
	}
```

Reply:
I will modify it. BTW, why it is dangerous ?

Reply:
I had modified this.